storage/storage.go

```diff
@@ -345,11 +346,35 @@ func constructWhereClauseForContent(reportRules types.ReportRules) string {
 	return statement
 }
 
+func getExtraDataFromReportRules(rules []types.RuleContentResponse, reportRules types.ReportRules) []types.RuleContentResponse {
+	if len(reportRules.HitRules) == 0 {
+		return rules
+	}
+
+	for _, ruleContent := range rules {
+		module := ruleContent.RuleModule
+		//errorKey := ruleContent.ErrorKey
+
+		for _, hitRule := range reportRules.HitRules {
```

Comment:
this N**2 loop can be a point for optimizations in the future. Now it's totally fine though since we don't have many rules.

Reply:
Yes, this code is ***, + there is the same loop in the constructing the where in, so I was thinking about just putting the details in a map during that time and later just retrieve from variable.

 and especially since we are extremely time limited